src/main/java/com/crosswordservice/controller/CrosswordController.java

```diff
@@ -96,19 +98,33 @@ public List<Question> updateAllQuestions(@RequestBody List<Question> questions,
 
     /**
      * Deletes a question with the given id.
+     *
      * @param id Id of a question
      * @return deleted question
      */
     @DeleteMapping("/questions/{id}")
-    public Question removeQuestion(@PathVariable Long id){
-        Question question = questionRepository.getReferenceById(id);
-        if(question == null){
+    public Optional<Question> removeQuestion(@PathVariable Long id){
+        if(!questionRepository.existsById(id)){
             throw new ResponseStatusException(HttpStatus.NOT_FOUND,"question not found");
         }
+        Optional<Question> question = questionRepository.findById(id);
+
         questionRepository.deleteById(id);
         return question;
```

Comment:
Why do you return an optional? Either the object with the ID does not exist and you get a 404 error message back or it is deleted and you get the deleted object back.